Src/GoogleApis.Auth.DotNet4/OAuth2/ServiceAccountCredential.cs

```diff
@@ -66,12 +70,27 @@ public Initializer(string id)
                 : this(id, GoogleAuthConsts.TokenUrl) { }
 
             /// <summary>Constructs a new initializer using the given id and the token server URL.</summary>
-            public Initializer(string id, string tokenServerUrl) : base(tokenServerUrl)
+            public Initializer(string id, string tokenServerUrl)
+                : base(tokenServerUrl)
             {
                 Id = id;
                 Scopes = new List<string>();
             }
 
+            /// <summary>Constructs a new initializer using the given id and the private key in the ClientCredentialParameters.</summary>
+            public Initializer(ClientCredentialParameters clientCredentialParameters)

```

Comment:
Given we have method FromCertificate(X509Certificate2) to create service account credential from the old format of credential obtained from developer console, why don't we have a method   FromJsonKey(...)  that would initialize the credential using the new recommended format of the key?

Actually, because it looks like this review is becoming a pretty long thread,
what if we just split it in two and first delivered the PR with support for reading service account credentials from JSON key (that would bring the bouncy castles dependency and just a tiny change to the API).

After we have that, we can proceed with ApplicationDefaultCredentials.